src/main/java/org/assertj/core/api/ClassBasedNavigableIterableAssert.java

```diff
@@ -53,7 +53,7 @@ public ELEMENT_ASSERT toAssert(ELEMENT value, String description) {
       throw new RuntimeException("Fail to find a constructor matching " + value
                                  + " class to build the expected Assert class");
     } catch (Exception e) {
-      throw new RuntimeException("Fail to build an assert object with " + value);
+      throw new RuntimeException("Fail to build an assert object with " + value + ": " + e.getMessage());
```

Comment:
I guess it would be even more helpful to add the whole exception instead of just the message? (e.g. `new RuntimeException("Failed to build an assert object with " + value, e)`)